xdsl/irdl/constraints.py

```diff
@@ -72,12 +72,17 @@ def get_unique_base(self) -> type[Attribute] | None:
         """Get the unique base type that can satisfy the constraint, if any."""
         return None
 
+    def __or__(
+        self, value: GenericAttrConstraint[AttributeInvT], /
+    ) -> AnyOf[AttributeCovT | AttributeInvT]:
```

Comment:
```suggestion
    def __or__(
        self, value: GenericAttrConstraint[AttributeCovT2], /
    ) -> AnyOf[AttributeCovT | AttributeCovT2]:
```
Both arguments should be covariant here I believe

Reply:
I'll give this a go